onadata/apps/logger/models/xform.py

```diff
@@ -1415,6 +1415,27 @@ def save_project(sender, instance=None, created=False, **kwargs):
 pre_save.connect(save_project, sender=XForm, dispatch_uid="save_project_xform")
 
 
+def _create_meta_perms(sender, instance, created, **kwargs):
+    metadata = instance.metadata_set.all().filter(data_type="xform_meta_perms").first()
+    if instance.deleted_by is None and metadata is None:
```

Comment:
Should we care whether the object is deleted or not? I think we can omit the check.